### PR TITLE
feat: infinite scroll room discovery

### PR DESCRIPTION
**Pull request by Aldiwildan77**
# Description
Add infinite scroll on room discovery

## Screenshot/Screen Recording

https://github.com/user-attachments/assets/a27d7326-df6b-4bff-ba3a-f5ec1bbcf568

## Release Notes

- **New Features**
	- Introduced infinite scrolling for room search results, allowing users to load more results as they scroll.
	- Updated the room card display with a new image source for improved visual representation.
	- Added a new dependency to enhance functionality.

- **Bug Fixes**
	- Adjusted default room limit in search results from 10 to 5 for improved performance.

- **Navigation Update**
	- Changed the navigation link for 'discover room' to direct users to the new search page.

### CODE REVIEW

File: src/modules/room/RoomSearchContainer.tsx [hunk 1]
```diff
@@ -34,10 +35,27 @@ export function RoomSearchContainer() {
...
+  useEffect(() => {
+    if (
+      inView &&
+      searchRoomQuery.hasNextPage &&
+      !searchRoomQuery.isFetchingNextPage
+    ) {
+      searchRoomQuery.fetchNextPage();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [inView, searchRoomQuery.hasNextPage, searchRoomQuery.isFetchingNextPage]);
```
[the remark this answers is not in the document]
Comment (Aldiwildan77) [reply src/modules/room/RoomSearchContainer.tsx]: updated @coderabbitai

File: src/modules/room/RoomSearchContainer.tsx [hunk 2]
```diff
@@ -83,28 +106,35 @@ export function RoomSearchContainer() {
...
+              {/* LOADING */}
               {searchRoomQuery.isLoading && <Spinner />}
-              {/* success with data */}
+
+              {/* DATA */}
               {!searchRoomQuery.isLoading &&
-                searchRoomQuery.data &&
-                searchRoomQuery.data?.rooms.length > 0 && (
+                searchRoomQuery.data?.pages &&
+                searchRoomQuery.data.pages.length > 0 && (
```
Comment (Fakhri1999) [inline src/modules/room/RoomSearchContainer.tsx:118]: is the `?.pages` come from the `useInfiniteQuery` hooks?

Comment (Aldiwildan77) [reply src/modules/room/RoomSearchContainer.tsx:118]: yes @Fakhri1999 u can check it on https://tanstack.com/query/latest/docs/framework/react/guides/infinite-queries

Comment (Fakhri1999) [reply src/modules/room/RoomSearchContainer.tsx:118]: oohh okayy nicee

**Review by Fakhri1999 — APPROVED** [review]

**Pull request merged**